bitcoin/src/pow.rs

```diff
@@ -225,6 +225,7 @@ impl Target {
     /// [target]: crate::blockdata::block::Header::target
     #[cfg_attr(all(test, mutate), mutate)]
     pub fn difficulty(&self, network: Network) -> u128 {
+        debug_assert_ne!(self.0, U256::ZERO);
```

Comment:
In 14ab447e37411cde144eb81bad10ba5b425dcf3f:

So, we already have an `assert` (not even a `debug_assert`) in `U256::div_rem`. So this additional assertion provides no value.

I think that if we want to add an assertion, it should be at the point where we compute a 0 target, not at the point where we try to compute its difficulty.

Reply:
Oops, in 7163eb91c5cd6d5aa66ededdcb95c8cd9f9eccf3 I mean.

Reply:
Woops I've force pushed and not fixed this. Fixing now.

Reply:
Oh, the value it adds is we get an assertion failure inside the `difficulty` function in debug builds, with the intention that this is easier to debug. That was the whole reason we put it here instead of panic'ing (you are however correct, and I missed, that the panic comes from our code not stdlib).

Reply:
I left it in and added a code comment.